src/main/java/br/com/metricminer2/domain/Commit.java

```diff
@@ -100,12 +101,17 @@ public Calendar getDate() {
 		return date;
 	}
 
-	public void addBranch(String branch) {
+	public void addBranch(String branch, String masterBranch) {
```

Comment:
Great! Since the `GitRepository` is the one that constructs `Commit`, instead of a `private GitRepository.setBranches()` and a `Commit.addBranch()`, I changed to a `getBranches()`, looked up for the masterBranch in hashes and then pass the boolean+branches to the `Commit` constructor all at once!